### DISCUSSION TITLE
Programatic MenuItem

### DISCUSSION THREAD

**Discussion by chrishj59** [primefaces/community#190] [PrimeReact] [answered]
I am using Appolo.  I would like to have a dynamic menu based on the content held in the DB.
I am using NextJs 13.5.4 using the the old pages directory

I am calling it with the menu with:
`return <AppSubMenu model={model2} />`

If I use the below, I get the error: Uncaught TypeError: props.model.map is not a function

`const model2 = (): MenuModel[] => {
		return [
			{
				label: 'Lebel 1',
			},
		];
	};`

But this works:
`const model1: MenuModel[] = [
		{
			label: 'Lebel 1',
		},
	];`

**Comment by habubey** [comment]
Hi @chrishj59 , we will check and get back to you. Thank you for your feedback

**Reply by chrishj59** [reply, marked as answer]
Issue was a basic one react one not calling the function
